### PR TITLE
update protobuf version

### PR DESCRIPTION
**Pull request by young-zy**

### CODE REVIEW

File: common/serial/typed_message.go [hunk 1]
```diff
@@ -21,16 +23,17 @@ func ToTypedMessage(message proto.Message) *TypedMessage {
...
 func GetMessageType(message proto.Message) string {
-	return proto.MessageName(message)
+	return string(message.ProtoReflect().Descriptor().FullName())
 }
 
 // GetInstance creates a new instance of the message with messageType.
 func GetInstance(messageType string) (interface{}, error) {
-	mType := proto.MessageType(messageType)
-	if mType == nil || mType.Elem() == nil {
+	log.Println(messageType)
```
Comment (Loyalsoldier) [inline common/serial/typed_message.go:31]: Is this line needed?

**Comment by Loyalsoldier** [comment]
It seems that this is NOT an incompatible change, so I suggest merge it into master branch.

**Comment by young-zy** [comment]
> It seems that this is NOT an incompatible change, so I suggest merge it into master branch.

seems v5 brach removed infra/control which uses some of the removed functions`MarshalText` and `UnmarshalText`. 

https://github.com/v2fly/v2ray-core/blob/291eff1aa347e275735520541d051a42cf0f2dce/infra/control/api.go#L99-L155

Also, it seems that google didn't ship replacements for original ones.

**Comment by Loyalsoldier** [comment]
Do you mean that this change is hard to be imported to master branch? If so, then just apply to v5.

**Comment by Loyalsoldier** [comment]
The file `proxy/vless/encoding/addons.proto` uses `github.com/gogo/protobuf/protoc-gen-gofast` to generate `proxy/vless/encoding/addons.pb.go` file.

If to change to package `google.golang.org/protobuf/proto`, build failed:

```
# v2ray.com/core/proxy/vless/encoding
proxy/vless/encoding/addons.go:33:41: cannot use addons (type *Addons) as type protoreflect.ProtoMessage in argument to "google.golang.org/protobuf/proto".Unmarshal:
	*Addons does not implement protoreflect.ProtoMessage (missing ProtoReflect method)
```

**Comment by young-zy** [comment]
> The file `proxy/vless/encoding/addons.proto` uses `github.com/gogo/protobuf/protoc-gen-gofast` to generate `proxy/vless/encoding/addons.pb.go` file.
> 
> If to change to package `google.golang.org/protobuf/proto`, build failed:
> 
> ```
> # v2ray.com/core/proxy/vless/encoding
> proxy/vless/encoding/addons.go:33:41: cannot use addons (type *Addons) as type protoreflect.ProtoMessage in argument to "google.golang.org/protobuf/proto".Unmarshal:
> 	*Addons does not implement protoreflect.ProtoMessage (missing ProtoReflect method)
> ```
> 
> You can run `go generate proto.go` in project root directory to reproduce this. If to keep using `protoc-gen-gofast` for it, can it be fixed?

`google.golang.org/protobuf/proto` should be compatible with package `golang/protobuf` version greater than 1.4.x

changing the import of `addons.go` back to `github.com/golang/protobuf/proto` should be a feasible solution. 

Though the build could pass the tests, since there seems to be few of tests on vless protocol, I cannot fully guarantee the functionalities. 

FYI: 
As far as I know, the latest released version of  `gogo/protobuf` has not yet supported `golang/protobuf release 1.4.x` (that is API v2 i'm trying to migrate to). At the same time, the future of `gogo/protobuf` cannot be guaranteed. for more specific details, click [here](https://github.com/gogo/protobuf/issues/691).

**Pull request merged**